### PR TITLE
Adding ATP assignments for PaaS Databases

### PR DESCRIPTION
**Pull request by Springstone**
## Overview/Summary

Added two new policy initiative assignments to enable Advanced Threat Detection for databases at intermediate root:
  - [Configure Advanced Threat Protection to be enabled on open-source relational databases](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/e77fc0b3-f7e9-4c58-bc13-cb753ed8e46e.html)
  - [Configure Azure Defender to be enabled on SQL Servers and SQL Managed Instances](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/9cb3cc7a-b39b-4b82-bc89-e5a5d9ff7b97.html)

Addresses user story: [AB#24092](https://dev.azure.com/CSUSolEng/2a9d5a5a-8998-4ad0-81c8-ef3045e4da97/_workitems/edit/24092)

## This PR fixes/adds/changes/removes

1. Adds policy assignment: [Configure Advanced Threat Protection to be enabled on open-source relational databases](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/e77fc0b3-f7e9-4c58-bc13-cb753ed8e46e.html)
2. Adds policy assignment:  [Configure Azure Defender to be enabled on SQL Servers and SQL Managed Instances](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/9cb3cc7a-b39b-4b82-bc89-e5a5d9ff7b97.html)

### Breaking Changes

1. N/A

## Testing Evidence

Please provide any testing evidence to show that your Pull Request works/fixes as described and planned (include screenshots, if appropriate).

#### Azure Public

[Deploy To Azure](https://portal.azure.com/#view/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2FSpringstone%2FEnterprise-Scale%2Fpolicy-refresh-mdfc%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2FSpringstone%2FEnterprise-Scale%2Fpolicy-refresh-mdfc%2FeslzArm%2Feslz-portal.json)

## As part of this Pull Request I have

- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [X] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [X] Performed testing and provided evidence.
- [X] Ensured [contribution guidance](https://github.com/Azure/Enterprise-Scale/wiki/ALZ-Contribution-Guide) is followed.
- [X] Updated relevant and associated documentation.
- [X] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

File: eslzArm/eslzArm.json [hunk 1]
```diff
@@ -1631,6 +1635,60 @@
...
+                "templateLink": {
+                    "contentVersion": "1.0.0.0",
+                    "uri": "[variables('deploymentUris').atpOssDbPolicyInitiative]"
+                },
+                "parameters": {
+                    "topLevelManagementGroupPrefix": {
+                        "value": "[parameters('enterpriseScaleCompanyPrefix')]"
+                    },
+                    "enforcementMode": {
+                                    "value": "Default"
```
Comment (jtracey93) [inline eslzArm/eslzArm.json:1660]: excessive indent. Also why arent we doing the normal enforcementMode logic to make it audit if folks want to?

File: eslzArm/eslzArm.json [hunk 2]
```diff
@@ -1631,6 +1635,60 @@
...
+                "templateLink": {
+                    "contentVersion": "1.0.0.0",
+                    "uri": "[variables('deploymentUris').atpSqlDbPolicyInitiative]"
+                },
+                "parameters": {
+                    "topLevelManagementGroupPrefix": {
+                        "value": "[parameters('enterpriseScaleCompanyPrefix')]"
+                    },
+                    "enforcementMode": {
+                                    "value": "Default"
```
Comment (jtracey93) [inline eslzArm/eslzArm.json:1687]: excessive indent. Also why arent we doing the normal enforcementMode logic to make it audit if folks want to?

File: eslzArm/managementGroupTemplates/policyAssignments/DINE-AtpOssDbPolicyAssignment.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            ],
+            "defaultValue": "Default"
+        }
+    },
+    "variables": {
+        "policyDefinitions": {
+            "DineAtpOssDb": "/providers/Microsoft.Authorization/policySetDefinitions/e77fc0b3-f7e9-4c58-bc13-cb753ed8e46e"
+        },
+        "policyAssignmentNames": {
+            "DineAtpOssDb": "Dine-Atp-Oss-Db",
```
Comment (jtracey93) [inline eslzArm/managementGroupTemplates/policyAssignments/DINE-AtpOssDbPolicyAssignment.json:25]:
```suggestion
            "DineAtpOssDb": "Deploy-MDFC-OssDb",
```

File: eslzArm/managementGroupTemplates/policyAssignments/DINE-AtpSqlDbPolicyAssignment.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    "variables": {
+        "policyDefinitions": {
+            "DineAtpSqlDb": "/providers/Microsoft.Authorization/policySetDefinitions/9cb3cc7a-b39b-4b82-bc89-e5a5d9ff7b97"
+        },
+        "policyAssignmentNames": {
+            "DineAtpSqlDb": "Dine-Atp-Sql-Db",
+            "description": "Enable Azure Defender on your SQL Servers and SQL Managed Instances to detect anomalous activities indicating unusual and potentially harmful attempts to access or exploit databases.",
+            "displayName": "Configure Azure Defender to be enabled on SQL Servers and SQL Managed Instances"
+        },
+        "rbacSqlSecurityManager": "	056cd41c-7e88-42e1-933e-88ba6a50c9c3",
```
Comment (jtracey93) [inline eslzArm/managementGroupTemplates/policyAssignments/DINE-AtpSqlDbPolicyAssignment.json:29]:
```suggestion
        "rbacSqlSecurityManager": "056cd41c-7e88-42e1-933e-88ba6a50c9c3",
```

File: eslzArm/managementGroupTemplates/policyAssignments/DINE-AtpSqlDbPolicyAssignment.json [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            ],
+            "defaultValue": "Default"
+        }
+    },
+    "variables": {
+        "policyDefinitions": {
+            "DineAtpSqlDb": "/providers/Microsoft.Authorization/policySetDefinitions/9cb3cc7a-b39b-4b82-bc89-e5a5d9ff7b97"
+        },
+        "policyAssignmentNames": {
+            "DineAtpSqlDb": "Dine-Atp-Sql-Db",
```
Comment (jtracey93) [inline eslzArm/managementGroupTemplates/policyAssignments/DINE-AtpSqlDbPolicyAssignment.json:25]:
```suggestion
            "DineAtpSqlDb": "Deploy-MDFC-SqlAtp",
```

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Thanks @Springstone, a few bits.

Also why arent we exposing in the portal?

File: eslzArm/eslzArm.json (hunk 1 above)
Comment (Springstone) [reply eslzArm/eslzArm.json:1660]: Blonde...

**Comment by Springstone** [comment]
> 

I'm using the existing switches in the portal:
image

File: eslzArm/eslzArm.json (hunk 2 above)
Comment (Springstone) [reply eslzArm/eslzArm.json]: Changed to use enforcement/auditing, instead of not provisioning is "no" selected.

**Pull request merged**